librz/analysis/arch_profile.c

```diff
@@ -11,38 +11,58 @@ RZ_API void rz_arch_profile_free(RzArchProfile *p) {
 	if (!p) {
 		return;
 	}
-	free(p->arch);
-	free(p->cpu);
 	ht_up_free(p->registers_mmio);
 	ht_up_free(p->registers_extended);
 }
 
 RZ_API RZ_OWN RzArchProfile *rz_arch_profile_new() {
-	RzArchProfile *cpu = RZ_NEW0(RzArchProfile);
-	if (!cpu) {
+	RzArchProfile *profile = RZ_NEW0(RzArchProfile);
+	if (!profile) {
 		return NULL;
 	}
-	cpu->db = sdb_new0();
-	cpu->arch = NULL;
-	cpu->cpu = NULL;
+	profile->ROM_SIZE = 0;
+	profile->RAM_SIZE = 0;
+	profile->EEPROM_SIZE = 0;
+	profile->IO_SIZE = 0;
+	profile->SRAM_START = 0;
+	profile->SRAM_SIZE = 0;
+	profile->PC = 0;
+	profile->PAGE_SIZE = 0;
+	profile->INTERRUPT_VECTOR_SIZE = 0;
```

Comment:
i would avoid to all use uppercase names, because usually we refer them as macros. in the code, but if you set them all lowercase, they will look perfect as they are.

Reply:
Yeah, sure, I'll get this done now.